### PR TITLE
New errors.md docs

### PR DESCRIPTION
**Pull request by ewels**
Addresses new features in `nf-core/tools` added in nf-core/tools#33 and nf-core/tools#38

### CODE REVIEW

File: errors.md [hunk 1]
```diff
@@ -148,11 +148,40 @@ if they are set.
...
-* Each dependency listed must have a version number pinned, eg. `toolname=1.6.8`
 
 Each dependency is checked using the [Anaconda API service](https://api.anaconda.org/docs).
-Dependency sublists (eg. `- pip`) are ignored.
-Dependency-specific channels (eg. `conda-forge::openjdk`) are ok.
+Dependency sublists are ignored with the exception of `- pip`: these packages are also checked
+for pinned version numbers and checked using the [PyPI JSON API](https://wiki.python.org/moin/PyPIJSON).
 
-* (Test failure) The package cannot be found on any of the listed channels
+Note that conda dependency with pinned channels (eg. `conda-forge::openjdk`) are fine
```
Comment (apeltzer) [inline errors.md]: dependency -> dependencies

File: errors.md [hunk 2]
```diff
@@ -148,11 +148,40 @@ if they are set.
...
+and should be handled by the linting properly.
+
+Each dependency can have the following lint failures and warnings:
+
+* (Test failure) Dependency does not version number pinned, eg. `toolname=1.6.8`
```
Comment (apeltzer) [inline errors.md]: * (Test failure) Dependency does not version number pinned, eg. `toolname=1.6.8` 

Should be (?)

 * (Test failure) Dependency does not match version number pinned, eg. `toolname=1.6.8`

**Review by apeltzer — CHANGES_REQUESTED** [review]
Two very minor syntax/typos... otherwise good to go :+1:

File: errors.md [hunk 3]
```diff
@@ -148,11 +148,40 @@ if they are set.
...
+RUN conda update -n base conda && \
+    conda env create -f /environment.yml && \
+    conda clean -a
+ENV PATH /opt/conda/envs/nfcore-EXAMPLE/bin:$PATH
+```
+
+This test only runs if there is both `environment.yml` and `Dockerfile` present in the workflow.
+
+Failures are generated if the `FROM`, `COPY`, `RUN` and `ENV` statements above are not present.
+Note that these lines must be an exact copy of the above example, with the exception that
```
Comment (maxulysse) [inline errors.md]: We're saying that `Such Dockerfiles can usually be very minimal...` and then that `that these lines must be an exact copy of the above example`.
I feel like there is some contradiction there

File: errors.md (hunk 2 above)
Comment (ewels) [reply errors.md]: I think I meant "does not _**have a**_ version number pinned"

27cscr

File: errors.md (hunk 3 above)
Comment (ewels) [reply errors.md]: I think that they can be both minimal and an exact copy, but I get where you're coming from.. I'll have a play 👍

File: errors.md [hunk 4]
```diff
@@ -148,11 +148,45 @@ if they are set.
...
+    description="Docker image containing all requirements for the nf-core/EXAMPLE pipeline"
+
+COPY environment.yml /
+RUN conda update -n base conda && \
+    conda env create -f /environment.yml && \
+    conda clean -a
+ENV PATH /opt/conda/envs/nfcore-EXAMPLE/bin:$PATH
+```
+
+To enforce this minimal `Dockerfile` and check for common copy+paste errors, we require
```
Comment (maxulysse) [inline errors.md:186]: Toppen

**Review by apeltzer — APPROVED** [review]

**Review by apeltzer — APPROVED** [review]
Looks all good now - merging!

**Pull request merged**